src/providers/README.md

```diff
@@ -2,3 +2,85 @@
 
 This directory contains providers contributing additional bindings, for example
 custom sequence actions.
+
+## Overview
+
+A provider is a class that provides a `value()` function. This function is called `Context` when another entity requests a value to be injected.
+
+Here we create a provider for a logging function that can be used in a custom sequence action.
```

Comment:
"used as a custom sequence action" or perhaps "used as a new action in a custom sequence".